src/memory_region/raw.rs

```diff
@@ -0,0 +1,80 @@
+use super::MrAccess;
+use crate::protection_domain::ProtectionDomain;
+use crate::LocalMrAccess;
+use rdma_sys::{ibv_access_flags, ibv_dereg_mr, ibv_mr, ibv_reg_mr};
+use std::fmt::Debug;
+use std::io;
+use std::{ptr::NonNull, sync::Arc};
+use utilities::{mut_ptr_to_usize, Cast};
+
+/// Raw Memory Region
+pub(crate) struct RawMemoryRegion {
+    /// the internal `ibv_mr` pointer
+    inner_mr: NonNull<ibv_mr>,
+    /// the addr of the raw memory region
+    addr: *mut u8,
+    /// the len of the raw memory region
+    len: usize,
+    /// the protection domain the raw memory region belongs to
+    _pd: Arc<ProtectionDomain>,
+}
+
+impl MrAccess for RawMemoryRegion {
+    fn addr(&self) -> usize {
+        mut_ptr_to_usize(self.addr)
+    }
+
+    fn length(&self) -> usize {
+        self.len
+    }
+
+    fn rkey(&self) -> u32 {
+        unsafe { *self.inner_mr.as_ptr() }.rkey
+    }
+}
+
+impl LocalMrAccess for RawMemoryRegion {
+    fn lkey(&self) -> u32 {
+        unsafe { self.inner_mr.as_ref() }.lkey
+    }
+}
+
+impl RawMemoryRegion {
+    /// Register a raw memory region from protection domain
+    #[allow(clippy::restriction)]
```

Comment:
Everytime we allow a clippy rule we should provide a good reason and comment it.